tlatools/org.lamport.tlatools/src/tlc2/tool/liveness/TableauNodePtrTable.java

```diff
@@ -130,19 +132,33 @@ public final long get(final long k, final int tidx) {
 		}
 	}
 
+	public final void put(final long k, final int tidx) {
+		// Writing (long) UNDONE will result in node[3] = -2,
+		// and DONE will be node[3] = -3.  isDone is
+		// true if node[3] = -2.  Method setDone also sets
+		// node[3] = - 3 to mark the node as done.
+		put(k, tidx, UNDONE, DONE);

```

Comment:
Is this call really guaranteed to modify `node[3]`? It seems like it should be possible that the node for fingerprint `k` already contains
 1. header
 2. data for tidx A
 3. data for tidx B

In which case a call to `put(k, B)` would leave `node[3]` unchanged. Is such a scenario possible?

Reply:
For better or worse, the call is guaranteed to modify `node[3]`. The following (simplified) test fails before and after the bugfix commit:

```java
	@Test
	public void testFoo() {
		
		final TableauNodePtrTable tbl = new TableauNodePtrTable(0);
		
		final long fp = 42L;
		final int t = 1;
		
		tbl.put(fp, t, 23);
		assertTrue(tbl.isDone(fp));
		assertEquals(23, tbl.get(fp, t));
		
		tbl.setDone(fp);
		assertTrue(tbl.isDone(fp));
		
		assertEquals(23, tbl.get(fp, t));
		
		tbl.put(fp, t);
		assertEquals(23, tbl.get(fp, t)); \\ Violated
		assertTrue(tbl.isDone(fp));  \\ Violated
	}
```

Any call in `LiveCheck` to `put(fp, t)` is guarded by a check if `getPtr(fp, t)` equals `-1`.  Thus, `put(fp, t)` should not destroy the state of `fp` in TNPT.